### PR TITLE
[5131] Site Explorer: pagination is not working

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/5131

### CODE REVIEW

File: ui/app/src/components/PathNavigator/PathNavigator.tsx [hunk 1]
```diff
@@ -158,7 +159,7 @@ export function PathNavigator(props: PathNavigatorProps) {
   const { id: siteId, uuid } = useActiveSite();
   const user = useActiveUser();
   const { authoringBase } = useEnv();
-  const dispatch = useDispatch();
+  const dispatch = useDispatch();@ro
```
Comment (rart) [inline ui/app/src/components/PathNavigator/PathNavigator.tsx]: `@ro`? 👆 

Test your build!

File: ui/app/src/state/epics/pathNavigator.ts [hunk 2]
```diff
@@ -209,7 +210,7 @@ export default [
           fetchChildrenByPath(state.sites.active, state.pathNavigator[id].currentPath, {
             limit: state.pathNavigator[id].limit,
             ...(Boolean(state.pathNavigator[id].keyword) && { keyword: state.pathNavigator[id].keyword }),
-            offset
+            offset: offset
```
Comment (rart) [inline ui/app/src/state/epics/pathNavigator.ts]: - Change doesn't seem to have a point. Revert?

File: ui/app/src/state/epics/pathNavigator.ts [hunk 3]
```diff
@@ -209,7 +210,7 @@ export default [
           fetchChildrenByPath(state.sites.active, state.pathNavigator[id].currentPath, {
             limit: state.pathNavigator[id].limit,
```
Comment (rart) [inline ui/app/src/state/epics/pathNavigator.ts]: You're not retrieving the limit from the action's payload but the offset is. Please double check.

File: ui/app/src/state/reducers/pathNavigator.ts [hunk 4]
```diff
@@ -250,6 +251,10 @@ const reducer = createReducer<LookupTable<PathNavigatorStateProps>>(
       ...state,
       [id]: { ...state[id], isFetching: true }
     }),
+    [pathNavigatorChangeLimit.type]: (state, { payload: { id, limit } }) => ({
+      ...state,
+      [id]: { ...state[id], limit, isFetching: true }
+    }),
```
Comment (rart) [inline ui/app/src/state/reducers/pathNavigator.ts:257]: You're also sending offset on the action, shouldn't that be updated too?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/PathNavigator/PathNavigator.tsx (hunk 1 above)
Comment (aachavarria) [reply ui/app/src/components/PathNavigator/PathNavigator.tsx]: I did not send the fix, it was on my commits

File: ui/app/src/state/reducers/pathNavigator.ts (hunk 4 above)
Comment (aachavarria) [reply ui/app/src/state/reducers/pathNavigator.ts:257]: no, the offset is being updated on the complete

**Review by rart — APPROVED** [review]

**Pull request merged**